sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/IServiceBusSessionManager.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.messaging.servicebus.implementation.DispositionStatus;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+
+/**
+ * A TEMPORARY interface describing the contract to work with one (V1, V2) or multiple sessions (V1).
+ * Note: Once the library is completely on V2 stack (i.e, when side-by-side V1 support is no longer needed),
+ * we will delete the V1 type {@link ServiceBusSessionManager} and this interface, so excuse the 'I' in the name
+ * of this "TO-BE-DELETED" interface .
+ */
+interface IServiceBusSessionManager extends AutoCloseable {
```

Comment:
Err.. I think this name is very C#-ish. Lol. 😆 I like the apology comment up there.

Reply:
I knew you will notice it given C# is your first love :). So I apologized early as this is a _temporary_ type added to keep the ServiceBusReceiverAsyncClient diff minimal for v1-v2 side by side support. 

Here was the plan, once we’re completely on V2 -

1.	ServiceBusReceiverAsyncClient that is obtained from the ServiceBusSessionReceiverAsyncClient which deals with **Single-Session** will use "ServiceBusSessionReactorReceiver" internally.
2.	The public Session-Processor abstraction that deals with **Multiple-Sessions** will use V2 SessionMessagePump, so Session-Processor has no dependency on ServiceBusReceiverAsyncClient.

This means, this _temporary_ interface "IServiceBusSessionManager", "V1 ServiceBusSessionManager" and _temporary_ "V2 ServiceBusSingleSessionMaanger" will be deleted. 

There won’t be anything useful in our "V1 ServiceBusSessionManager" as we completely rewrote the multi-session processing with "V2 SessionMessagePump".